src/app/helpers/lazyImageLoading.js

```diff
@@ -0,0 +1,33 @@
+const createLazyListener = (selector, dataset) => () => {
+  let images = document.querySelectorAll(selector);
+  if (images.length) {
+    images = [].slice.call(images);
+    setTimeout(() => {
+      images.forEach(lazyImage => {
+        const { top, bottom } = lazyImage.getBoundingClientRect();
+        const topIsUnderScreenOrHigher = top - 100 <= window.innerHeight;
+        const botIsOnScreenOrBelow = bottom >= 0;
+        const visible = lazyImage.style.display !== 'none';
+
+        if (topIsUnderScreenOrHigher && botIsOnScreenOrBelow && visible) {
+          lazyImage.src = lazyImage.dataset.src;
+
+          setTimeout(() => {
+            delete lazyImage.dataset[dataset];
+          }, 200);
+        }
+      });
+    }, 0);
+  }
+};
+
+export default function lazyImageLoading(selector, dataset, parent = window) {
+  const listener = createLazyListener(selector, dataset);
+  listener();
+  parent.addEventListener('scroll', listener);
```

Comment:
when I'm reading this 3 lines I have strange feeling that it's not well designed )
Could you short explain what is happening there?

Reply:
I create listener, I call it, because I need to load images that are already on the screen before I scroll, then I add event listener:)

Reply:
> when I'm reading this 3 lines I have strange feeling that it's not well designed )
> Could you short explain what is happening there?

I create listener, I call it, because I need to load images that are already on the screen before I scroll, then I add event listener:)